src/FulfillerMultiSelect.jsx

```diff
@@ -100,39 +117,73 @@ class FulfillerMultiSelect extends React.Component {
     }
 
     handleChange(e) {
-        const fulfillerIds = e && e.map(f => f.fulfiller ? f.fulfiller.fulfillerId : f.fulfillerId);
-        if (!e || fulfillerIds === this.state.selectedFulfillerIds) {
+        if (!e) {
             return;
         }
 
+        const fulfillerIds = this.props.multi
+            ? e.map(f => f.fulfiller.fulfillerId)
+            : [e.fulfiller.fulfillerId];
+
         this.setState({
             selectedFulfillerIds: fulfillerIds
         });
 
+        this.onSelectedFulfillerChanged(fulfillerIds);
+    }
+
+    onSelectedFulfillerChanged(fulfillerIds, updateCustomizrSettings = true) {
         if (this.props.onChange) {
-            this.props.onChange(fulfillerIds.map(f => this.fulfillerMap[f]));
+            this.props.onChange(this.mapFulfillerIdsToFulfillers(fulfillerIds));
         }
 
-        this.updateRecentFulfillerIds(fulfillerIds);
+        if (updateCustomizrSettings) {
+            this.updateRecentFulfillerIds(fulfillerIds);
+        }
     }
 
     async getRecentFulfillerIds() {
         if (this.props.accessToken) {
             let settings = await this.customizrClient.getSettings(this.props.accessToken);
-            let recentFulfillerIds = settings.recentFulfillerIds || [];
-            this.setState({ recentFulfillerIds });
-            return recentFulfillerIds;
+            return {
+                recentFulfillerIds: settings.recentFulfillerIds || [],
+                recentFulfillerIdsMulti: settings.recentFulfillerIdsMulti || []
+            };
         }
     }
 
     async updateRecentFulfillerIds(fulfillerIds) {
+        if (!this.props.multi
```

Comment:
For multi select we want to store every selection because the `recentFulfillerIdsMulti` would change